Setting Teams Meeting Policy settings when AllowCloudRecording is False returns error

ISSUE TITLE:
Setting Teams Meeting Policy settings when AllowCloudRecording is False returns error

ISSUE DESCRIPTION (this template):
Changing any settings when AllowCloudRecording  is set to False (including setting AllowCloudRecording to False) returns error below:


##[error]Updating the recording storage is not allowed when AllowCloudRecording is set to false.
    + CategoryInfo          : OperationStopped: (:) [], CimException
    + FullyQualifiedErrorId : AggregateException,Microsoft.Rtc.Management.Internal.SetTeamsMeetingPolicyCmdlet
    + PSComputerName        : localhost


#### The operating system the target node is running
Windows Server 2019

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.21.331.1

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_TeamsMeetingPolicy/MSFT_TeamsMeetingPolicy.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_TeamsMeetingPolicy/MSFT_TeamsMeetingPolicy.psm1
@@ -531,6 +531,11 @@ function Set-TargetResource
         {
             $SetParameters.Remove("AllowAnonymousUsersToDialOut") | Out-Null
         }
+        if ($SetParameters.AllowCloudRecording -eq $false )
+        {
+            $SetParameters.Remove("RecordingStorageMode")
+            $SetParameters.Remove("AllowRecordingStorageOutsideRegion")
+        }
         Set-CsTeamsMeetingPolicy @SetParameters
     }
     elseif ($Ensure -eq 'Absent' -and $CurrentValues.Ensure -eq 'Present')

